crates/uv-build-frontend/src/lib.rs

```diff
@@ -517,17 +546,17 @@ impl SourceBuild {
                 let pyproject_toml: PyProjectToml =
                     PyProjectToml::deserialize(pyproject_toml.into_deserializer())
                         .map_err(Error::InvalidPyprojectTomlSchema)?;
+                let name = pyproject_toml
+                    .project
+                    .as_ref()
+                    .map(|project| &project.name)
+                    .or(package_name);
 
                 let backend = if let Some(build_system) = pyproject_toml.build_system {
                     // If necessary, lower the requirements.
                     let requirements = match source_strategy {
                         SourceStrategy::Enabled => {
-                            if let Some(name) = pyproject_toml
-                                .project
-                                .as_ref()
-                                .map(|project| &project.name)
-                                .or(package_name)
-                            {
+                            if let Some(name) = name {
```

Comment:
There's some unreverted noise in this file from Aria's change. I'll clean it up.